website_sale_tax_toggle/controllers/main.py

```diff
@@ -8,10 +9,38 @@
 
 
 class WebsiteSaleTaxToggle(WebsiteSale):
+    @http.route()
+    def shop(
+        self,
+        page=0,
+        category=None,
+        search="",
+        min_price=0.0,
+        max_price=0.0,
+        ppg=False,
+        **post
+    ):
+        res = super().shop(
+            page=page,
+            category=category,
+            search=search,
+            min_price=min_price,
+            max_price=max_price,
+            ppg=ppg,
+            **post
+        )
+        default_tax_toggle = request.website.default_tax_toggle
+        if request.session.get("tax_toggle_taxed") is None:
+            request.session["tax_toggle_taxed"] = default_tax_toggle
+        return res
+
     @http.route(["/website/tax_toggle"], type="json", auth="public", website=True)
     def tax_toggle(self):
-        # Create a session variable
-        request.session["tax_toggle_taxed"] = not request.session.get(
-            "tax_toggle_taxed", False
-        )
+        default_tax_toggle = request.website.default_tax_toggle
+        if request.session.get("tax_toggle_taxed") is None:
+            request.session["tax_toggle_taxed"] = default_tax_toggle
+        else:
+            request.session["tax_toggle_taxed"] = not request.session[
+                "tax_toggle_taxed"
+            ]
```

Comment:
We have to make the switch anyway:

```suggestion
        request.session["tax_toggle_taxed"] = not request.session[
            "tax_toggle_taxed"
        ]
```